packages/cli/templates/react/src/App.tsx

```diff
@@ -1,24 +1,21 @@
-import { useFieldPlugin } from './useFieldPlugin'
 import FieldPlugin from './components/FieldPlugin'
 import { FunctionComponent } from 'react'
+import { FieldPluginProvider } from './FieldPluginProvider'
 
 const App: FunctionComponent = () => {
-  const { type, data, actions } = useFieldPlugin()
-
-  if (type === 'loading') {
-    return <span>Loading...</span>
-  }
-
-  if (type === 'error') {
-    return <span>Error</span>
-  }
-
-  return type === 'loaded' ? (
-    <FieldPlugin
-      data={data}
-      actions={actions}
-    />
-  ) : null
+  return (
+    <FieldPluginProvider
+      Loading={Loading}
+      Error={Error}
+    >
+      <FieldPlugin />
+    </FieldPluginProvider>
+  )
 }
 
+const Loading: FunctionComponent = () => <p>Loading...</p>
+const Error: FunctionComponent<{ error: Error }> = (props) => {
+  console.error(props.error)
+  return <p>An error occured, please see the console for more details.</p>
+}
```

Comment:
@eunjae-lee I extracted these components to preserve referential equality.

I am also logging the error message. This will be useful when the plugin is opened outside of the sandbox, as the error message states the details of what happens and how to resolve it: the message contains a link to the container app/sandbox.

<img width="484" alt="image" src="https://user-images.githubusercontent.com/14206504/232000625-a1256065-93f7-49a4-a66f-104b98eaa268.png">
<img width="603" alt="image" src="https://user-images.githubusercontent.com/14206504/232000674-8580aebb-f997-4873-bf0f-3e1d4d1db82d.png">